File: GameEngine_PC/GameEngineOpenGL3_2/src/main/java/gameEngine/models/AudioSource.java
package gameEngine.models;

/**
 * Source represents the object that can play sounds for instance the CD player
 */
public class AudioSource {

	/**
	 * Identifier of the source in openAL
	 */
	private int sourceId;
	
	/**
	 * Reference to the buffer currently in the source if any
	 */
	private AudioBuffer buffer;

	/**
	 * @param sourceId
	 *            Identifier of the source in openAL
	 */
	public AudioSource(int sourceId) {
		super();
		this.sourceId = sourceId;
	}

	/**
	 * @return the sourceId Identifier of the source in openAL
	 */
	public int getSourceId() {
		return sourceId;
	}

	/**
	 * @param sourceId
	 *            Identifier of the source in openAL to set
	 */
	public void setSourceId(int sourceId) {
		this.sourceId = sourceId;
	}

	/**
	 * @return the buffer currently associated with source
	 */
	public AudioBuffer getBuffer() {
		return buffer;
	}

	/**
	 * @param buffer the buffer to set in the source
	 */
	public void setBuffer(AudioBuffer buffer) {
		this.buffer = buffer;
	}

	

}

File: GameEngine_IOS/GameEngine/AudioEngine/Models/AudioSource.swift
import Foundation

import OpenAL


/// Source represents the object that can play sounds for instance the CD player
open class AudioSource {
    

    /// Identifier of the source in openAL
    fileprivate var sourceId : ALuint;
    
    /// Reference to the buffer currently in the source if any
    fileprivate var buffer : AudioBuffer!;
    
    /// Initializer of the audio source
    ///
    /// - Parameter sourceId: Identifier of the source in openAL
    public init(sourceId : ALuint) {
        self.sourceId = sourceId;
        self.buffer = nil;
    }
    
    /// Getter of the source identifier
    ///
    /// - Returns: The sourceId Identifier of the source in openAL
    open func getSourceId() -> ALuint {
    return sourceId;
    }
    
    
    /// Getter of the associated buffer
    ///
    /// - Returns: The buffer currently associated with source
    open func getBuffer() -> AudioBuffer! {
        return self.buffer;
    }
    

    /// Associated a buffer to the source
    ///
    /// - Parameter buffer: <#buffer description#>
    open func set(buffer : AudioBuffer!) {
        self.buffer = buffer;
    }
    
    
    
}
